cli/xgotext/parser/domain.go

```diff
@@ -26,13 +26,17 @@ func (t *Translation) AddLocations(locations []string) {
 }
 
 // Dump translation as string
-func (t *Translation) Dump() string {
-	data := make([]string, 0, len(t.SourceLocations)+5)
-
-	locations := t.SourceLocations
-	sort.Strings(locations)
-	for _, location := range locations {
-		data = append(data, "#: "+location)
+func (t *Translation) Dump(locations bool) string {
```

Comment:
We can't change this public interface. We need to either make the `locations` param optional, like `Dump(locations ...bool)` and handle it properly only for the first slice item, or add a new function , so the flag functionality gets splitted into these, like `Dump()` and `DumpLocales` or whatever. 

Same applies for the following `Idem` comments below